### PR TITLE
OCPQE-19032 add release-qe-test step and job

### PR DESCRIPTION
**Pull request by rioliu-rh**
- add new step called `release-qe-tests` to trigger job controller
- add job `qe-release-gate-test` to ref new step

### CODE REVIEW

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

File: ci-operator/config/openshift/release-tests/openshift-release-tests-master.yaml [hunk 1]
```diff
@@ -18,7 +22,7 @@ images:
   from: root
   to: oar-python3
 - dockerfile_path: prow/Dockerfile
-  from: tools
+  from: cli
```
Comment (jianzhangbjz) [inline ci-operator/config/openshift/release-tests/openshift-release-tests-master.yaml:25]: As we synced before, we'd better not change the old one, but build a new image.

File: ci-operator/step-registry/release-qe-tests/release-qe-tests-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+ref:
+  as: release-qe-tests
+  from: job-python3
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/release-qe-tests/release-qe-tests-ref.yaml:3]: We can integrate the `oc` here, like 
```
cli: latest
```

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

File: ci-operator/step-registry/release-qe-tests/release-qe-tests-ref.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  cli: "4.14"
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/release-qe-tests/release-qe-tests-ref.yaml:4]: How about `latest`?

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

File: ci-operator/step-registry/release-qe-tests/release-qe-tests-commands.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+for release in $VALID_RELEASES
+do
+  echo "start job controller for $release"
+  jobctl start-controller -r $release
+  jobctl start-controller -r $release --no-nightly
+done
+
+jobctl start-aggregator
+
+echo $RELEASE_PAYLOAD_MODIFIER_TOKEN
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/release-qe-tests/release-qe-tests-commands.sh:29]: We'd better not output it.

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Comment by rioliu-rh** [comment]
@jianzhangbjz new token works, can you review again

File: ci-operator/step-registry/release-qe-tests/release-qe-tests-commands.sh [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+github_token=$(cat "/var/run/vault/tests-private-account/token-git")
+export GITHUB_TOKEN=$github_token
+
+release_payload_modifier_token=$(cat /var/run/vault/release-payload-modifier-token/token)
+export RELEASE_PAYLOAD_MODIFIER_TOKEN=$release_payload_modifier_token
+
+echo "Login cluster app.ci"
+oc login api.ci.l2s4.p1.openshiftapps.com:6443 --token=$RELEASE_PAYLOAD_MODIFIER_TOKEN
+
```
Comment (jianzhangbjz) [inline ci-operator/step-registry/release-qe-tests/release-qe-tests-commands.sh:18]: Can we get the `releasepayload` resource successfully?

Comment (rioliu-rh) [reply ci-operator/step-registry/release-qe-tests/release-qe-tests-commands.sh:18]: yes, refer to this job https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/48939/rehearse-48939-periodic-ci-openshift-release-tests-master-qe-release-gate-test/1760306315902586880/artifacts/qe-release-gate-test/release-qe-tests/build-log.txt

```console
Login cluster app.ci
Logged into "https://api.ci.l2s4.p1.openshiftapps.com:6443" as "system:serviceaccount:ci:release-payload-modifier" using the token provided.

You have access to the following projects and can switch between them with 'oc project <projectname>':

  * edge-infrastructure
    fcos
    ocp
    ocp-arm64
    ocp-multi
    ocp-ppc64le
    ocp-s390x
    origin
    origin-web-console

Using project "edge-infrastructure".
system:serviceaccount:ci:release-payload-modifier
Test whehter releasepayload can be accessed
NAME                                 AGE
4.16.0-0.nightly-2024-02-20-014802   36h
```

**Comment by rioliu-rh** [comment]
/pj-rehearse periodic-ci-openshift-release-tests-master-qe-release-gate-test

**Pull request merged**